public/app/core/components/RolePicker/RolePickerMenu.tsx

```diff
@@ -0,0 +1,585 @@
+import React, { FormEvent, useCallback, useEffect, useState } from 'react';
+import { css, cx } from '@emotion/css';
+import {
+  Button,
+  Checkbox,
+  CustomScrollbar,
+  HorizontalGroup,
+  RadioButtonGroup,
+  Tooltip,
+  useStyles2,
+  useTheme2,
+} from '@grafana/ui';
+import { GrafanaTheme2, SelectableValue } from '@grafana/data';
+import { getSelectStyles } from '@grafana/ui/src/components/Select/getSelectStyles';
+import { OrgRole, Role } from 'app/types';
+
+type BuiltInRoles = { [key: string]: Role[] };
+
+const BuiltinRoles = ['Viewer', 'Editor', 'Admin'];
+const BuiltinRoleOption: Array<SelectableValue<OrgRole>> = BuiltinRoles.map((r: string) => ({
+  label: r,
+  value: r as OrgRole,
+}));
+
+const fixedRoleGroupNames: { [key: string]: string } = {
+  ldap: 'LDAP',
+  current: 'Current org',
+};
+
+interface RolePickerMenuProps {
+  builtInRole: OrgRole;
+  builtInRoles: BuiltInRoles;
+  options: Role[];
+  appliedRoles: Role[];
+  showGroups?: boolean;
+  onSelect: (roles: Role[]) => void;
+  onBuiltInRoleSelect?: (role: OrgRole) => void;
+  onUpdate: (newBuiltInRole: OrgRole, newRoles: string[]) => void;
+  onClear?: () => void;
+}
+
+export const RolePickerMenu = ({
+  builtInRole,
+  builtInRoles,
+  options,
+  appliedRoles,
+  showGroups,
+  onSelect,
+  onBuiltInRoleSelect,
+  onUpdate,
+  onClear,
+}: RolePickerMenuProps): JSX.Element => {
+  const [selectedOptions, setSelectedOptions] = useState<Role[]>(appliedRoles);
+  const [selectedBuiltInRole, setSelectedBuiltInRole] = useState<OrgRole>(builtInRole);
+  const [selectedBuiltInRoles, setSelectedBuiltInRoles] = useState<Role[]>(builtInRoles[builtInRole]);
+  const [showSubMenu, setShowSubMenu] = useState(false);
+  const [openedMenuGroup, setOpenedMenuGroup] = useState('');
+  const [subMenuOptions, setSubMenuOptions] = useState<Role[]>([]);
+
+  const theme = useTheme2();
+  const styles = getSelectStyles(theme);
+  const customStyles = useStyles2(getStyles);
+
+  const getOptionGroups = useCallback(() => {
+    const groupsMap: { [key: string]: Role[] } = {};
+    options.forEach((role) => {
+      if (role.name.startsWith('fixed:')) {
+        const groupName = getRoleGroup(role);
+        if (groupsMap[groupName]) {
+          groupsMap[groupName].push(role);
+        } else {
+          groupsMap[groupName] = [role];
+        }
+      }
+    });
+
+    const groups = [];
+    for (const groupName in groupsMap) {
+      if (Object.prototype.hasOwnProperty.call(groupsMap, groupName)) {
+        const groupOptions = groupsMap[groupName].sort(sortRolesByName);
+        groups.push({
+          name: fixedRoleGroupNames[groupName] || capitalize(groupName),
+          value: groupName,
+          options: groupOptions,
+        });
+      }
+    }
+    return groups.sort((a, b) => (a.name < b.name ? -1 : 1));
```

Comment:
For sorting strings [localeCompare](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare#sort_an_array) is a better option.